extensions/amp-analytics/0.1/requests.js

```diff
@@ -325,6 +369,25 @@ export class RequestHandler {
       this.refreshBatchInterval_();
     }, interval);
   }
+
+  /**
+   * Composes a request URL given a base and requestOrigin
+   * @private
+   * @param {string} baseUrl
+   * @param {string=} opt_requestOrigin
+   * @return {string}
+   */
+  composeRequestUrl_(baseUrl, opt_requestOrigin) {
+    if (opt_requestOrigin) {
+      // parse can handle relative request origins and will use the
+      // current page's origin if so. However, we only want to accept absolute
```

Comment:
This gives me the impression that we will be introducing a check for absolute url, which we are not. Can we say we expect the requestOrigin to always contains the url origin. In the case where it is a relative url, the current page's origin will be used.